src/Common/tests/System/PlatformDetection.cs

```diff
@@ -207,49 +207,72 @@ private static bool IsDistroAndVersion(string distroId, string versionId = null)
             return false;
         }
 
-        private static IdVersionPair ParseOsReleaseFile()
+        public static string GetDistroVersionString()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return "";
+            }
+
+            DistroInfo v = ParseOsReleaseFile();
+
+            return "Distro=" + v.Id + " VersionId=" + v.VersionId + " Pretty=" + v.PrettyName + " Version=" + v.Version;
+        }
+
+        private static DistroInfo ParseOsReleaseFile()
```

Comment:
It certainly needs tidying but that would involve updating 180 project files and adding Unix configurations to most of them?